meson.build

```diff
@@ -47,6 +47,12 @@ project('picolibc', 'c',
 targets = []
 
 cc = meson.get_compiler('c')
+fs = import('fs')
+host_cpu_family = host_machine.cpu_family()
+if host_cpu_family == 'riscv32' or host_cpu_family == 'riscv64'
```

Comment:
if we do this way, that must be a separate PR. `host_cpu_family` is used in various places and directory renaming doesn't fit with this "TLS" PR.

Reply:
that sounds good -- "fixing" the amd64 directory name would be a good plan anyways.

I was inspired by your great meson cleanups to go and "flatten" the build -- avoiding all of the intermediate library bits (except where necessary to have different c_args). I don't think that has a huge overlap with this series, but did want you to watch out in case that lands first.

Reply:
OK.  Then, this PR should be ready to merge. I'll make an issue about the arch dir thing. I'll work on the github speed up PR and the Zephyr port.  Thank you for all your reviews and comments!